jvb/src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java

```diff
@@ -68,22 +69,25 @@
      * A debug data structure which tracks outstanding buffers and tracks from where (via
      * a stack trace) they were requested and returned.
      */
-    private static final Map<Integer, String> bookkeeping
-            = new ConcurrentHashMap<>();
+    private static final Map<byte[], Exception> bookkeeping
+            = Collections.synchronizedMap(new IdentityHashMap<>());
+
+    private static final Map<byte[], Queue<BufferEvent>> bufferEvents =
+        Collections.synchronizedMap(new IdentityHashMap<>());
 
     private static class ReturnedBufferBookkeepingInfo
     {
-        final String allocTrace;
-        final String deallocTrace;
-        ReturnedBufferBookkeepingInfo(String a, String d)
+        final Exception allocTrace;
+        final Exception deallocTrace;
+        ReturnedBufferBookkeepingInfo(Exception a, Exception d)
         {
             allocTrace = a;
             deallocTrace = d;
         }
     }
 
-    private static final Map<Integer, ReturnedBufferBookkeepingInfo> returnedBookkeeping
-        = new ConcurrentHashMap<>();
+    private static final Map<byte[], ReturnedBufferBookkeepingInfo> returnedBookkeeping
+        = Collections.synchronizedMap(new IdentityHashMap<>());
```

Comment:
It looks like you're now just using the returnedBookkeeping as a Set to remember what's been returned - does it need to be a map at all?  I think you could do 
```
private static final Set<byte[]> returnedBookkeeping = Collections.synchronizedSet(Collections.newSetFromMap(new IdentityHashMap<>()));
```

And eliminate `ReturnedBufferBookkeepingInfo`.

Reply:
Yeah I think I can do the same thing for `bookkeeping` as well.